vision_agent/tools/tools.py

```diff
@@ -457,6 +457,42 @@ def __call__(self, input: List[int]) -> float:
         return round(input[0] / input[1], 2)
 
 
+class ExtractFrames(Tool):
+    name = "extract_frames_"
+    description = "'extract_frames_' extract image frames from the input video, return a list of tuple (frame, timestamp), where the timestamp is the relative time in seconds of the frame occurred in the video, the frame is a local image file path that stores the frame."
+    usage = {
+        "required_parameters": [{"name": "video_uri", "type": "str"}],
+        "examples": [
+            {
+                "scenario": "Can you extract the frames from this video? Video: www.foobar.com/video?name=test.mp4",
+                "parameters": {"video_uri": "www.foobar.com/video?name=test.mp4"},
+            },
+            {
+                "scenario": "Can you extract the images from this video file? Video path: tests/data/test.mp4",
+                "parameters": {"video_uri": "tests/data/test.mp4"},
+            },
+        ],
+    }
+
+    def __call__(self, video_uri: str) -> list[tuple[str, float]]:
+        try:
+            from vision_agent.tools.video import extract_frames_from_video
+        except Exception as e:
+            raise ImportError(
+                "vision_agent is not installed correctly (cause: missing dependencies), please run 'pip install vision-agent[video]' instead."
+            ) from e
+        frames = extract_frames_from_video(video_uri)
+        result = []
+        _LOGGER.info(
```

Comment:
more of an FYI I've been using the _LOGGER.info for printing in verbose mode, but not sure if that's the best way to do it (maybe debug would have been better?)

Reply:
ah, it depends on how verbose we want to be.
Some third party library could be very verbose and flood our logs if we allow debug level.
but we could also selectively turn on debug mode for certain modules. 
Anyway, i think we can tune these settings later down the road.